code/datums/wires/wires.dm

```diff
@@ -283,6 +283,47 @@
 /datum/wires/proc/get_wire(color)
 	return colors[color]
 
+/**
+ * Get the name of the wire passed in.
+ * If wires can't be read, may either return something scrambled or nothing at all.
+ * A null return here means that there should be no name displayed at all.
+ *
+ * Arguments:
+ * * wire - a wire define.
+ */
+/datum/wires/proc/get_wire_name(mob/user, wire)
+	if(user.can_admin_interact())
+		// admins always get a free pass
+		return wire
+
+	var/can_probably_see_wires = FALSE
+	var/obj/item/held_item = user.get_active_hand()
+	if(istype(held_item) && HAS_TRAIT(held_item, TRAIT_SHOW_WIRE_INFO))
+		can_probably_see_wires = TRUE
+	if(HAS_TRAIT(user, TRAIT_SHOW_WIRE_INFO))
+		can_probably_see_wires = TRUE
+
+	if(!can_probably_see_wires)
+		return null
+
+	// even if you *think* you can see them, it's not guaranteed that you can.
+	if(HAS_TRAIT(holder, TRAIT_OBSCURED_WIRES))
+		var/list/fake_wire_chars = list("!", "@", "#", "$", "%", "^", "&", "*", "(", ")", "-", "?", "/")
+		var/fake_wire_name = ""
+
+		for(var/i in 1 to 5)
+			fake_wire_name += pick(fake_wire_chars)
+
+		return fake_wire_name
+
+	// even if you can see everything, duds should still be null
+	if(is_dud(wire))
+		return null
+	return wire
+
+	// can_see_wire_info(user) && !is_dud_color(color) ? get_wire(color) : null
```

Comment:
```suggestion
	// can_see_wire_info(user) && !is_dud_color(color) ? get_wire(color) : null
```

??

Reply:
oops, I pulled that down so I didn't forget the checks. I'll remove that